packages/core/test/functions.spec.ts

```diff
@@ -678,6 +678,98 @@ describe('Stylable functions (native, formatter and variable)', () => {
     });
 
     describe('diagnostics', () => {
+        describe('deprecation', () => {
+            it('should return a warning when used deprecated stMap', () => {
+                expectWarningsFromTransform(
+                    {
+                        entry: '/style.st.css',
+                        files: {
+                            '/style.st.css': {
+                                content: `
+                                :vars {
+                                    |color1: $stMap$(key1 red)|;
+                                }
+
+                            `,
+                            },
+                        },
+                    },
+                    [
+                        {
+                            message: processorWarnings.DEPRECATED_ST_FUNCTION_NAME(
+                                'stMap',
+                                'st-map'
+                            ),
+                            file: '/style.st.css',
+                            severity: 'info'
+                        },
+                    ]
+                );
+            });
+
+            it('should return a warning when used deprecated stArray', () => {
+                expectWarningsFromTransform(
+                    {
+                        entry: '/style.st.css',
+                        files: {
+                            '/style.st.css': {
+                                content: `
+                                :vars {
+                                    |color1: $stArray$(red, blue)|;
+                                }
+                            `,
+                            },
+                        },
+                    },
+                    [
+                        {
+                            message: processorWarnings.DEPRECATED_ST_FUNCTION_NAME(
+                                'stArray',
+                                'st-array'
+                            ),
+                            file: '/style.st.css',
+                            severity: 'info'
+                        },
+                    ]
+                );
+            });
+
+            it('should return a warning when used deprecated nested stArray or stMap', () => {
```

Comment:
`should return a warning when using a nested deprecated "stArray" or "stMap" syntax`